src/components/DropdownMenuItem/DropdownMenuItem.tsx

```diff
@@ -57,57 +58,55 @@ export interface Props {
  *
  * Dropdown menu item within `DropdownMenu`
  */
-export const DropdownMenuItem = React.forwardRef<HTMLLIElement, Props>(
-  (
-    {
-      align,
-      className,
-      href,
-      children,
-      variant,
-      onClick,
-      status,
-      target,
-      ...other
-    },
-    ref,
-  ) => {
-    const TagName = createTagName();
+export const DropdownMenuItem = ({
+  align,
+  className,
+  href,
+  children,
+  variant,
+  onClick,
+  status,
+  target,
+  ...other
+}: DropdownMenuItemProps) => {
+  const context = useContext(DropdownMenuContext);
+  const ref = (el: HTMLLIElement) => {
+    if (context && el && !context.refs.current.set.has(el)) {
+      context.refs.current.set.add(el);
+      context.refs.current.list.push(el);
+    }
```

Comment:
Is the `set` just to make sure you don't push a ref to the `list` more than once?

Reply:
Yea, some re-renders can cause `list` to expand with the same elements, so the set helps to prevent that

Reply:
Could you add a comment to that effect?